input/docs/handbook/data-binding/xamarin-forms.md

```diff
@@ -67,7 +79,43 @@ public partial class TheContentPage : ReactiveContentPage<TheViewModel>
 }
 ```
 
-# Routing
+The ReactiveUI binding engine is very powerful, but there may be cases where you want to bind to C# Events coming from your view (such as when a buttons `Tap` event fires). For this scenario use [Pharmacist](https://github.com/reactiveui/Pharmacist), which automatically creates Observables for all the C# Events in your project.
+
+```csharp
+public partial class TheContentPage : ReactiveContentPage<TheViewModel>
+{
+    public ThePage()
+    {
+        InitializeComponent();
+
+        // Setup the bindings.
+        // Note: We have to use WhenActivated here, since we need to dispose the
+        // bindings on XAML-based platforms, or else the bindings leak memory.
+        this.WhenActivated(disposable =>
+        {
+            this.Bind(ViewModel, x => x.TheText, x => x.TheTextBox.Text)
+                .DisposeWith(disposable);
+            this.OneWayBind(ViewModel, x => x.TheText, x => x.TheTextBlock.Text)
+                .DisposeWith(disposable);
+                
+            this.TheTextButton
+                //Provided by Pharmacist
+                .Events()
+                .Released
+                //Don't pass the EventArgs to the command
+                .Select(_ => Unit.Default)
+                .InvokeCommand(ViewModel, x => x.TheTextCommand)
+                .DisposeWith(disposable)p
+        });
+    }
+}
+```
+
+### Performance Tips
```

Comment:
On xamarin forms this often is counter to our recommendations. Eg disposing can have a performance cost and may not always be needed. 

Eg this is more about preventing memory leaks and there are scenarios when you don't need to dispose.

Reply:
Ah ok. So `WhenActivated` is more about providing an easy way to control the lifetime of a binding (only needs to be disposed when needed to improve performance), as well as preventing memory leaks?

Reply:
Kind of, `WhenActivated` and disposing observables isn't more performant, just more correct in certain situations. Eg on WPF/UWP you always have to Dispose due to a potential memory leak which was the motivation of introducing it.

Observable derived from events have the same lifetime rules as events. 

Eg if the current object is the owner of the memory then the GC can free the object. If the current object doesn't own it then the GC can't free the object.

Reply:
So for Xamarin.Forms what is the correct way to use `WhenActivated` 🙂

Reply:
https://reactiveui.net/docs/handbook/when-activated/#when-should-i-bother-disposing-of-idisposable-objects

Reply:
I've removed the performance tips that i wrote, i'll re-add my words when i've looked into the performance a bit more